src/arkworks_ext.rs

```diff
@@ -0,0 +1,44 @@
+use ark_ed_on_bls12_377::Fq;
+use ark_ff::SquareRootField;
+use ark_serialize::CanonicalSerialize;
+
+use crate::sign::Sign;
+
+pub trait FieldFormatExt: Sized {
+    fn to_decimal_string(&self) -> String;
+}
+
+pub trait XSqrt: Sized {
+    // This trait matches the xsqrt function in the sage spec,
+    // always returning a positive root if it exists.
+    fn xsqrt(&self) -> Option<Self>;
+}
```

Comment:
This should probably be (or be unified with, if the behavior is different) the `SqrtRatioZeta` trait defined in `invsqrt.rs`.

Reply:
One point that we should make sure of is that the specification of the elligator map should *not* make any assumptions about the sign of the value returned by the square root function.

If we need the square root to have a particular sign, we should compute an absolute value explicitly.  The reason is that sign checks are [very expensive](https://protocol.penumbra.zone/main/crypto/decaf377/costs.html#sign-checks) in the circuit, so we should be careful to only use them explicitly.

This might require a bit of care working through the spec description of the elligator map, rather than using the behavior of the `xsqrt` function.

Reply:
[this part](https://github.com/penumbra-zone/decaf377/blob/main/ristretto.sage#L563-L566) of the spec does requires the roots of n1 and n2 to be positive, then another sign change is applied such that s is positive when sqrt(n1) exists, otherwise s is negative, but I've modified the behavior on our side to use the absolute value as suggested